### PR TITLE
Hw 33

### PR DESCRIPTION
**Pull request by KrohIgor**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -101,6 +101,12 @@
             <artifactId>jackson-databind</artifactId>
             <version>2.11.0</version>
         </dependency>
+
```
Comment (dmytro-hryhoruk) [inline pom.xml]:
```suggestion
```

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 2]
```diff
@@ -14,9 +14,7 @@
 @Configuration
 @PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
-        "com.dev.cinema.dao",
-        "com.dev.cinema.security",
-        "com.dev.cinema.service"
+        "com.dev.cinema",
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/config/AppConfig.java]: let's add only those packages which we need to Scan.

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/cinemahalls")
+public class CinemaHallController {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]:
```suggestion
@RequestMapping(value = "/cinema-halls")
public class CinemaHallController {
```

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/moviesessions")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]:
```suggestion
@RequestMapping(value = "/movie-sessions")
```

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Autowired
+    public OrderController(OrderService orderService, ShoppingCartService shoppingCartService,
+                           UserService userService) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+    }
+
+    @PostMapping(value = "/complete")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/OrderController.java]:
```suggestion
    @PostMapping
```

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/shoppingcarts")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]:
```suggestion
@RequestMapping(value = "/shopping-carts")
```

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    @Autowired
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  MovieSessionService movieSessionService,
+                                  UserService userService) {
+        this.shoppingCartService = shoppingCartService;
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+    }
+
+    @PostMapping(value = "/addmoviesession")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java:35]:
```suggestion
    @PostMapping
```

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    public void addMovieSession(@RequestParam Long userId,
+                                @RequestBody ShoppingCartRequestDto shoppingCartRequestDto) {
+        shoppingCartService.addSession(movieSessionService.get(shoppingCartRequestDto
+                .getMovieSessionId()), userService.get(userId));
+    }
+
+    @GetMapping(value = "/byuser")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java:42]:
```suggestion
    @GetMapping(value = "/by-user")
```

File: src/main/java/com/dev/cinema/controller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+public class UserController {
+
+    private UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping(value = "/byemail")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/UserController.java:22]:
```suggestion
    @GetMapping(value = "/by-email")
```

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 10]
```diff
@@ -47,6 +50,20 @@ public CinemaHall add(CinemaHall cinemaHall) {
...
 
+    @Override
+    public CinemaHall get(Long cinemaHallId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<CinemaHall> criteriaQuery = criteriaBuilder.createQuery(CinemaHall.class);
+            Root<CinemaHall> root = criteriaQuery.from(CinemaHall.class);
+            Predicate predicate = criteriaBuilder.equal(root.get("id"), cinemaHallId);
+            criteriaQuery.where(predicate);
+            return session.createQuery(criteriaQuery).getSingleResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java:61]:
```suggestion
         return session.get(CinemaHall.class, cinemaHallId);
```
try this one

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 11]
```diff
@@ -46,6 +49,20 @@ public Movie add(Movie movie) {
...
 
+    @Override
+    public Movie get(Long movieId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Movie> criteriaQuery = criteriaBuilder.createQuery(Movie.class);
+            Root<Movie> root = criteriaQuery.from(Movie.class);
+            Predicate predicate = criteriaBuilder.equal(root.get("id"), movieId);
+            criteriaQuery.where(predicate);
+            return session.createQuery(criteriaQuery).getSingleResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java:60]: same

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 12]
```diff
@@ -49,6 +49,20 @@ public User add(User user) {
...
 
+    @Override
+    public User get(Long userId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
+            Root<User> root = criteriaQuery.from(User.class);
+            Predicate predicate = criteriaBuilder.equal(root.get("id"), userId);
+            criteriaQuery.where(predicate);
+            return session.createQuery(criteriaQuery).getSingleResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java:60]: )

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto;
+
+import java.time.LocalDateTime;
+
+public class MovieSessionRequestDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
+package com.dev.cinema.model.dto;
+
+import java.time.LocalDateTime;
+
+public class MovieSessionResponseDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/model/dto/OrderRequestDto.java [hunk 15]
```diff
@@ -0,0 +1,19 @@
+package com.dev.cinema.model.dto;
+
+public class OrderRequestDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/OrderRequestDto.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/model/dto/OrderResponseDto.java [hunk 16]
```diff
@@ -0,0 +1,50 @@
+package com.dev.cinema.model.dto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderResponseDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/OrderResponseDto.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/model/dto/ShoppingCartRequestDto.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
+package com.dev.cinema.model.dto;
+
+public class ShoppingCartRequestDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/ShoppingCartRequestDto.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/model/dto/ShoppingCartResponseDto.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/ShoppingCartResponseDto.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/model/dto/UserRequestDto.java [hunk 19]
```diff
@@ -0,0 +1,29 @@
+package com.dev.cinema.model.dto;
+
+public class UserRequestDto {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/dto/UserRequestDto.java]:
```suggestion
```
remove these empty line everywhere in your classes

**Review by dmytro-hryhoruk — COMMENTED** [review]
add class-mappers for Dtos

File: pom.xml (hunk 1 above)
Comment (KrohIgor) [reply pom.xml]: done

File: src/main/java/com/dev/cinema/config/AppConfig.java (hunk 2 above)
Comment (KrohIgor) [reply src/main/java/com/dev/cinema/config/AppConfig.java]: done

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java (hunk 3 above)
Comment (KrohIgor) [reply src/main/java/com/dev/cinema/controller/CinemaHallController.java]: done

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java (hunk 4 above)
Comment (KrohIgor) [reply src/main/java/com/dev/cinema/controller/MovieSessionController.java]: done

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 5 above)
Comment (KrohIgor) [reply src/main/java/com/dev/cinema/controller/OrderController.java]: done

File: src/main/java/com/dev/cinema/model/dto/UserRequestDto.java (hunk 19 above)
Comment (KrohIgor) [reply src/main/java/com/dev/cinema/model/dto/UserRequestDto.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**